CRM/MembersOnlyEvent/Hook/Copy/Event.php

```diff
@@ -19,10 +19,10 @@ public function handle($objectName, &$object) {
     if (!$this->shouldHandle($objectName)) {
       return;
     }
-    $this->createMemberOnlyEventFromEventTemplate($object);
+    $this->createMemberOnlyEventFromEventTemplate($object->id);
   }
 
-  protected function createMemberOnlyEventFromEventTemplate(&$objectRef) {
+  protected function createMemberOnlyEventFromEventTemplate($id) {
```

Comment:
this should be private unless you go with the option (1) I mentioned in my other comment then you can change it to public